src/main/java/org/prebid/server/bidder/tradplus/TradPlusBidder.java

```diff
@@ -0,0 +1,143 @@
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public class TradPlusBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpTradPlus>> EXT_TYPE_REFERENCE =
+            new TypeReference<>() {
+            };
+
+    private static final String ZONE_ID = "{{ZoneID}}";
+    private static final String ACCOUNT_ID = "{{AccountID}}";
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public TradPlusBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest bidRequest) {
+        final Map<ExtImpTradPlus, List<Imp>> extToImps = new HashMap<>();
+        try {
+            final ExtImpTradPlus extImpTradPlus = parseImpExt(bidRequest.getImp().getFirst().getExt());
+            validateImpExt(extImpTradPlus);
+            extToImps.computeIfAbsent(extImpTradPlus, ext -> new ArrayList<>()).add(bidRequest.getImp().getFirst());
+        } catch (PreBidException e) {
+            return Result.withError(BidderError.badInput(e.getMessage()));
+        }
+
+        final List<HttpRequest<BidRequest>> httpRequests = extToImps.entrySet().stream()
+                .map(entry -> makeHttpRequest(entry.getKey(), entry.getValue(), bidRequest))
+                .toList();
+
+        return Result.withValues(httpRequests);
+    }
+
+    private ExtImpTradPlus parseImpExt(ObjectNode extNode) {
+        final ExtImpTradPlus extImpTradPlus;
+        try {
+            extImpTradPlus = mapper.mapper().convertValue(extNode, EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage());
+        }
+        return extImpTradPlus;
+    }
+
+    private void validateImpExt(ExtImpTradPlus extImpTradPlus) {
+        if (StringUtils.isBlank(extImpTradPlus.getAccountId())) {
+            throw new PreBidException("Invalid/Missing AccountID");
+        }
+    }
+
+    private HttpRequest<BidRequest> makeHttpRequest(ExtImpTradPlus extImpTradPlus, List<Imp> imps,
+                                                    BidRequest bidRequest) {
+        final String uri = endpointUrl
+                .replace(ZONE_ID, extImpTradPlus.getZoneId())
+                .replace(ACCOUNT_ID, extImpTradPlus.getAccountId());
+
+        final BidRequest outgoingRequest = bidRequest.toBuilder().imp(removeFirstImpExt(imps)).build();
+
+        return BidderUtil.defaultRequest(outgoingRequest, uri, mapper);
+    }
+
+    private static List<Imp> removeFirstImpExt(List<Imp> imps) {
+        imps.set(0, imps.getFirst().toBuilder().ext(null).build());
+        return imps;
+    }
```

Comment:
this looks redundant since the list of imps always has a single imp

Reply:
I have changed these codes，please review these codes.